### PR TITLE
Add support for remote datasets (as git repositories)

### PR DESCRIPTION
**Pull request by jff**
This PR adds support for remote datasets (as git repositories) and configures Smartbugs with a new remote dataset: solidiFI.

SolidiFI is a dataset of buggy contracts injected with 9369 bugs of 7 different types: reentrancy, timestamp dependency, unhandled exceptions, unchecked send, TOD, integer overflow/underflow, and use of tx.origin.

### CODE REVIEW

File: smartBugs.py [hunk 1]
```diff
@@ -1,39 +1,42 @@
 #!/usr/bin/env python3
 
-import docker
+# import docker
 import yaml
-import logging
+# import logging
```
Comment (tdurieux) [inline smartBugs.py:5]: If not used, you can remove them

File: src/interface/cli.py [hunk 2]
```diff
@@ -1,15 +1,15 @@
-
 #!/usr/bin/env python3
 import yaml
-import logging
+# import logging
```
Comment (tdurieux) [inline src/interface/cli.py:3]: can be removed too

File: smartBugs.py (hunk 1 above)
Comment (ruimaranhao) [reply smartBugs.py:5]: +1

File: src/interface/cli.py (hunk 2 above)
Comment (ruimaranhao) [reply src/interface/cli.py:3]: +1

File: src/interface/cli.py [hunk 3]
```diff
@@ -52,13 +51,29 @@ def __call__(self, parser, namespace, values, option_string=None):
...
     remote_info = cfg_dataset[name]
     if isinstance(remote_info, list):
-        for prop in (e['url'] for e in remote_info if isinstance(e, dict) and 'url' in e):
+        merged = {}
+        for d in remote_info:
+            if isinstance(d, dict):
+                merged = merge_two_dicts(merged, d)
+
+        # A remote dataset needs to define an url and a local dir
+        if 'url' in merged and 'local_dir' in merged:
```
Comment (ruimaranhao) [inline src/interface/cli.py:65]: can this be simplified:

`return 'url' in merged and 'local_dir' in merged`

File: smartBugs.py [hunk 4]
```diff
@@ -85,7 +83,7 @@ def exec_cmd(args: argparse.Namespace):
                     # local copy does not exist; we need to clone it
                     print('\x1b[1;37m' + "%s is a remote dataset. Do you want to create a local copy? [Y/n] " % base_name + '\x1b[0m')
                     answer = input()
-                    if answer in ['y', 'Y', 'Yes', 'YES', 'Ye', 'YE']:
+                    if answer in ['y', 'Y', 'yes', 'Yes', 'YES', '']:
```
Comment (ruimaranhao) [inline smartBugs.py:86]: Does the empty string ('') make sense in this list?

Comment (jff) [reply smartBugs.py:86]: It represents the default option (yes). If you press enter, it's the same as yes.

Comment (ruimaranhao) [reply smartBugs.py:86]: Would it work if someone would write `yEs`? if so, perhaps the better is:

`answer.lower() in ['yes', 'y', '']`

Comment (jff) [reply smartBugs.py]: Done.

**Pull request merged**